src/hooks/useUser/useUser.tsx

```diff
@@ -0,0 +1,177 @@
+
+export const ActiveUserProvider: React.FC = ({ children }) => {
+  const { activeUserState, setActiveUser, resetActiveUser } = useActiveUserStore()
+
+  const [accounts, setAccounts] = useState<Account[] | null>(null)
+  const [extensionConnected, setExtensionConnected] = useState<boolean | null>(null)
+
+  const accountsIds = (accounts || []).map((a) => a.id)
+  const {
+    memberships: membershipsData,
+    previousData: membershipPreviousData,
+    loading: membershipsLoading,
+    error: membershipsError,
+    refetch: refetchMemberships,
+  } = useMemberships({ where: { controllerAccount_in: accountsIds } }, { skip: !accounts || !accounts.length })
+
+  // use previous values when doing the refetch, so the app doesn't think we don't have any memberships
+  const memberships = membershipsData || membershipPreviousData?.memberships
+
+  const {
+    membership: activeMembership,
+    loading: activeMembershipLoading,
+    error: activeMembershipError,
+    refetch: refetchActiveMembership,
+  } = useMembership({ where: { id: activeUserState.memberId } }, { skip: !activeUserState.memberId })
+
+  if (membershipsError) {
+    throw membershipsError
+  }
+
+  if (activeMembershipError) {
+    throw activeMembershipError
+  }
+
+  // handle polkadot extension
+  useEffect(() => {
+    let unsub: () => void
+
+    const initPolkadotExtension = async () => {
+      try {
+        const enabledExtensions = await web3Enable(WEB3_APP_NAME)
+
+        if (!enabledExtensions.length) {
+          console.warn('No Polkadot extension detected')
+          setExtensionConnected(false)
+          return
+        }
+
+        const handleAccountsChange = (accounts: InjectedAccountWithMeta[]) => {
+          const mappedAccounts = accounts.map((a) => ({
+            id: a.address,
+            name: a.meta.name || 'Unnamed',
+          }))
+          setAccounts(mappedAccounts)
+        }
+
+        // subscribe to changes to the accounts list
+        unsub = await web3AccountsSubscribe(handleAccountsChange)
+        const accounts = await web3Accounts()
+        handleAccountsChange(accounts)
+
+        setExtensionConnected(true)
+      } catch (e) {
+        setExtensionConnected(false)
+        console.error('Unknown polkadot extension error', e)
+      }
+    }
+
+    initPolkadotExtension()
+
+    return () => {
+      unsub?.()
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!accounts || !activeUserState.accountId || extensionConnected !== true) {
+      return
+    }
+
+    const account = accounts.find((a) => a.id === activeUserState.accountId)
+
+    if (!account) {
+      console.warn('Selected accountId not found in extension accounts, resetting user')
+      resetActiveUser()
+    }
+  }, [accounts, activeUserState.accountId, extensionConnected, resetActiveUser])
+
+  const userInitialized =
+    (extensionConnected === true && (!!memberships || !accounts?.length)) || extensionConnected === false
```

Comment:
```suggestion
const userInitialized = !!(extensionConnected && (!!memberships || !accounts?.length))
```
Should work, right? Not 100% sure

Reply:
I don't think it does work because `false` is a valid state for `extensionConnected` (user doesn't have the extension yet)